pkg/sql/logictest/testdata/logic_test/inflight_trace_spans

```diff
@@ -0,0 +1,53 @@
+# Verify that the crdb_internal.node_inflight_trace_spans vtable populates
+# correctly.
+
+statement ok
+GRANT ADMIN TO testuser
+
+statement ok
+CREATE TABLE kv (k VARCHAR PRIMARY KEY, v VARCHAR);
+
+query TT
+SELECT * FROM kv
+----
+
+user testuser
+
+statement ok
+BEGIN
+
+let $curr_trace_id
+SELECT * FROM crdb_internal.trace_id()
+
+# Save all rows representing spans of current trace in view.
+statement ok
+CREATE VIEW current_trace_spans(span_id, trace_id)
+  AS SELECT span_id, trace_id
+  FROM crdb_internal.node_inflight_trace_spans
+  WHERE trace_id = $curr_trace_id
+
+# Confirm that there is at least 1 trace with current trace ID.
+query B
+SELECT count(*) > 0
+  FROM current_trace_spans
+----
+true
+
+statement ok
+INSERT INTO kv VALUES('k', 'v');
+COMMIT
+
+# Confirm that the trace from earlier ended.
+query T
+SELECT trace_id
+  FROM crdb_internal.node_inflight_trace_spans
+  WHERE trace_id = $curr_trace_id
+----
+
+# Confirm that none of the spans from earlier trace are still in vtable.
+query T
+SELECT vtable.span_id
```

Comment:
I think this block can never return anything if the above block above goes through successfully.

Reply:
Yeah, I agree — I added it just in case we weren't sure that the span IDs were tied to trace IDs? @tbg mentioned saving a temp table (view) in #59873 and I figured he wanted me to check for all associated spans of the original trace.

Reply:
I think just asserting on the following should suffice for both these blocks:
```
SELECT count(*) > 0
   FROM current_trace_spans
 ----
false
```

Reply:
Oh yeah, that's smart 🤩 updated now!